models/issues/label.go

```diff
@@ -143,37 +144,32 @@ func (l *Label) CalOpenOrgIssues(ctx context.Context, repoID, labelID int64) {
 
 // LoadSelectedLabelsAfterClick calculates the set of selected labels when a label is clicked
 func (l *Label) LoadSelectedLabelsAfterClick(currentSelectedLabels []int64, currentSelectedExclusiveScopes []string) {
-	labelQuerySlice := []int64{}
+	labelQueryParams := container.Set[string]{}
 	labelSelected := false
-	labelScope := l.ExclusiveScope()
-	for i, s := range currentSelectedLabels {
-		if s == l.ID {
+	exclusiveScope := l.ExclusiveScope()
+	for i, curSel := range currentSelectedLabels {
+		if curSel == l.ID {
 			labelSelected = true
-		} else if -s == l.ID {
+		} else if -curSel == l.ID {
 			labelSelected = true
 			l.IsExcluded = true
-		} else if s != 0 {
+		} else if curSel != 0 {
 			// Exclude other labels in the same scope from selection
-			if s < 0 || labelScope == "" || labelScope != currentSelectedExclusiveScopes[i] {
-				labelQuerySlice = append(labelQuerySlice, s)
+			if curSel < 0 || exclusiveScope == "" || exclusiveScope != currentSelectedExclusiveScopes[i] {
+				labelQueryParams.Add(strconv.FormatInt(curSel, 10))
 			}
 		}
 	}
 
 	if !labelSelected {
-		labelQuerySlice = append(labelQuerySlice, l.ID)
+		labelQueryParams.Add(strconv.FormatInt(l.ID, 10))
 	}
 	l.IsSelected = labelSelected
 
 	// Sort and deduplicate the ids to avoid the crawlers asking for the
 	// same thing with simply a different order of parameters
-	slices.Sort(labelQuerySlice)
-	labelQuerySlice = slices.Compact(labelQuerySlice)
-	// Quick conversion (strings.Join() doesn't accept slices of Int64)
-	labelQuerySliceStrings := make([]string, len(labelQuerySlice))
-	for i, x := range labelQuerySlice {
-		labelQuerySliceStrings[i] = strconv.FormatInt(x, 10)
-	}
+	labelQuerySliceStrings := labelQueryParams.Values()
+	slices.Sort(labelQuerySliceStrings) // the sort is still needed because the underlying map of Set doesn't guarantee order
```

Comment:
Little warning: probably acceptable in this case but you're sorting strings now so it will output something like `1,10,2` instead of `1,2,10`.

Reply:
Don't really see the necessity.

They are IDs and are totally transparent to other modules, the only thing we need to do is to make them ordered, so string comparing or number comparing doesn't matter.

Think about another case, if we have a lot of UUIDs as IDs, we could also just sort them, and we do not really care about why `A<B` or `A>b`